client/src/base_rules.js

```diff
@@ -738,6 +738,28 @@ export const ChessRules = class ChessRules {
     return moves;
   }
 
+  // Similar to getSlideNJumpMoves, but limited range
+  // TODO: DRY or combine with getSlideNJumpMoves
+  getSlideNJumpMovesLimited([x, y], steps, maxRange) {
```

Comment:
This can definitely be combined with getSlideNJumpMoves. (Same for  isAttackedBy...) Some probably-obvious thoughts here -- can hash this out if you're interested in moving forward with this